components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/internal/ServiceComponent.java

```diff
@@ -198,6 +214,124 @@ public Response saveExecutionPlan(String saveConfig) {
     }
 
     @GET
+    @Path("/workspace/root")
+    @Produces("application/json")
+    public Response root() {
+        try {
+            return Response.status(Response.Status.OK)
+                    .entity(workspace.listRoots())
+                    .header("Access-Control-Allow-Origin", '*')
+                    .type(MediaType.APPLICATION_JSON)
+                    .build();
+        } catch (Throwable ignored) {
+        }
+        return Response.serverError().entity("failed")
+                .header("Access-Control-Allow-Origin", "*")
+                .build();
+    }
+
+    @GET
+    @Path("/workspace/list")
+    @Produces("application/json")
+    public Response directoriesInPath(@QueryParam("path") String path) {
+        try {
+            return Response.status(Response.Status.OK)
+                    .entity(workspace.listDirectoriesInPath(new String(Base64.getDecoder().decode(path))))
+                    .header("Access-Control-Allow-Origin", '*')
+                    .type(MediaType.APPLICATION_JSON)
+                    .build();
+        } catch (Throwable ignored) {
+        }
+        return Response.serverError().entity("failed")
+                .header("Access-Control-Allow-Origin", "*")
+                .build();
+    }
+
+    @GET
+    @Path("/workspace/exists")
+    @Produces("application/json")
+    public Response pathExists(@QueryParam("path") String path) {
+        try {
+            return Response.status(Response.Status.OK)
+                    .entity(workspace.exists(new String(Base64.getDecoder().decode(path))))
+                    .header("Access-Control-Allow-Origin", '*')
+                    .type(MediaType.APPLICATION_JSON)
+                    .build();
+        } catch (Throwable ignored) {
+        }
+        return Response.serverError().entity("failed")
+                .header("Access-Control-Allow-Origin", "*")
+                .build();
+    }
+
+    @GET
+    @Path("/workspace/listFiles")
+    @Produces("application/json")
+    public Response filesInPath(@QueryParam("path") String path) {
+        try {
+            return Response.status(Response.Status.OK)
+                    .entity(workspace.listFilesInPath(new String(Base64.getDecoder().decode(path))))
+                    .header("Access-Control-Allow-Origin", '*').type(MediaType.APPLICATION_JSON).build();
+        } catch (Throwable ignored) {
+        }
+        return Response.serverError().entity("failed")
+                .header("Access-Control-Allow-Origin", "*")
+                .build();
+
+    }
+
+    @POST
+    @Path("/workspace/write")
+    @Produces("application/json")
+    public Response write(String payload) {
+        try {
```

Comment:
Shall we catch if there are any checked exceptions thrown first and handle them with proper error messages to the client and then only catch the throwable. Does it make sense?